src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4876,6 +4876,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           case LookupInaccessible(sym, msg) => issue(AccessError(tree, sym, context, msg))
           case LookupNotFound               =>
             inEmptyPackage orElse lookupInRoot(name) match {
+              // ignore wildcard bodies inserted by the Java parser
+              case NoSymbol if context.unit.isJava && name == termNames.WILDCARD => tree
```

Comment:
they are also inserted as parameter values, i.e. `void foo(int x)` gets parsed to `def foo(x: Int = _): Unit`.

The change in the typer was the best place I could find. However this is also the first time I'm working with the compiler's internals so maybe there is a better, more generic place to add the wildcard-ignore.